providers/sfmesh/sfmesh.go

```diff
@@ -0,0 +1,850 @@
+		}
+
+		if len(endpoints) > 0 {
+			codePackage.Endpoints = &endpoints
+		}
+
+		codePackages = append(codePackages, codePackage)
+	}
+
+	serviceProperties := servicefabricmesh.ServiceResourceProperties{}
+	serviceProperties.OsType = servicefabricmesh.Linux
+	replicaCount := int32(1)
+	serviceProperties.ReplicaCount = &replicaCount
+	serviceProperties.CodePackages = &codePackages
+	service.ServiceResourceProperties = &serviceProperties
+	services = append(services, service)
+	properties.Services = &services
+
+	return meshApp, nil
+}
+
+func (p *SFMeshProvider) getMeshNetwork(pod *v1.Pod, meshApp servicefabricmesh.ApplicationResourceDescription, location string) servicefabricmesh.NetworkResourceDescription {
+	network := servicefabricmesh.NetworkResourceDescription{}
+	network.Name = meshApp.Name
+	network.Location = &location
+
+	networkProperties := servicefabricmesh.NetworkResourceProperties{}
+	addressPrefix := "10.0.0.4/22"
+	networkProperties.AddressPrefix = &addressPrefix
+
+	layers := []servicefabricmesh.Layer4IngressConfig{}
+
+	service := (*meshApp.Services)[0]
+
+	for _, codePackage := range *service.CodePackages {
+		for _, endpoint := range *codePackage.Endpoints {
+			layer := p.getLayer(&endpoint, *meshApp.Name, *service.Name)
+			layers = append(layers, layer)
+		}
+	}
+
+	ingressConfig := servicefabricmesh.IngressConfig{}
+	ingressConfig.Layer4 = &layers
+
+	networkProperties.IngressConfig = &ingressConfig
+	network.NetworkResourceProperties = &networkProperties
+
+	return network
+}
+
+func (p *SFMeshProvider) getLayer(endpoint *servicefabricmesh.EndpointProperties, appName string, serviceName string) servicefabricmesh.Layer4IngressConfig {
+	layer := servicefabricmesh.Layer4IngressConfig{}
+	name := *endpoint.Name + "Ingress"
+	layerName := &name
+	layer.Name = layerName
+	layer.PublicPort = endpoint.Port
+	layer.EndpointName = endpoint.Name
+	layer.ApplicationName = &appName
+	layer.ServiceName = &serviceName
+
+	return layer
+}
+
+func (p *SFMeshProvider) getEndpointFromContainerPort(port v1.ContainerPort) servicefabricmesh.EndpointProperties {
+	endpoint := servicefabricmesh.EndpointProperties{}
+	endpointName := strconv.Itoa(int(port.ContainerPort)) + "Listener"
+	endpoint.Name = &endpointName
+	endpoint.Port = &port.ContainerPort
+
+	return endpoint
+}
+
+// CreatePod accepts a Pod definition and creates a SF Mesh App.
+func (p *SFMeshProvider) CreatePod(pod *v1.Pod) error {
+	log.Printf("receive CreatePod %q\n", pod.Name)
+
+	meshApp, err := p.getMeshApplication(pod)
+	if err != nil {
+		return err
+	}
+
+	meshNetwork := p.getMeshNetwork(pod, meshApp, p.region)
+	_, err = p.networkClient.Create(context.Background(), p.resourceGroup, *meshNetwork.Name, meshNetwork)
+	if err != nil {
+		return err
+	}
+
+	networkName := *meshNetwork.Name
+	resourceID := "/subscriptions/" + os.Getenv("AZURE_SUBSCRIPTION_ID") + "/resourceGroups/" + p.resourceGroup + "/providers/Microsoft.ServiceFabricMesh/networks/" + networkName
```

Comment:
prefer to add a property in SFMeshProvider to store the subscription ID, just like the p.resourceGroup.

Reply:
done